sed/loader/flash/buffer_handler.py

```diff
@@ -79,160 +129,137 @@ def _schema_check(self) -> None:
                     "Please check the configuration file or set force_recreate to True.",
                 )
 
-    def _get_files_to_read(
-        self,
-        h5_paths: list[Path],
-        folder: Path,
-        prefix: str,
-        suffix: str,
-        force_recreate: bool,
-    ) -> None:
+    def _save_buffer_file(self, paths: dict[str, Path]) -> None:
         """
-        Determines the list of files to read and the corresponding buffer files to create.
+        Creates the electron and timed buffer files from the raw H5 file.
+        First the dataframe is accessed and forward filled in the non-electron channels.
+        Then the data types are set. For the electron dataframe, all values not in the electron
+        channels are dropped. For the timed dataframe, only the train and pulse channels are taken
+        and it pulse resolved (no longer electron resolved). Both are saved as parquet files.
 
         Args:
-            h5_paths (List[Path]): List of paths to H5 files.
-            folder (Path): Path to the folder for buffer files.
-            prefix (str): Prefix for buffer file names.
-            suffix (str): Suffix for buffer file names.
-            force_recreate (bool): Flag to force recreation of buffer files.
+            paths (dict[str, Path]): Dictionary containing the paths to the H5 and buffer files.
         """
-        # Getting the paths of the buffer files, with subfolder as buffer and no extension
-        self.buffer_paths = initialize_paths(
-            filenames=[h5_path.stem for h5_path in h5_paths],
-            folder=folder,
-            subfolder="buffer",
-            prefix=prefix,
-            suffix=suffix,
-            extension="",
-        )
-        # read only the files that do not exist or if force_recreate is True
-        files_to_read = [
-            force_recreate or not parquet_path.exists() for parquet_path in self.buffer_paths
-        ]
 
-        # Get the list of H5 files to read and the corresponding buffer files to create
-        self.missing_h5_files = list(compress(h5_paths, files_to_read))
-        self.save_paths = list(compress(self.buffer_paths, files_to_read))
+        # Create a DataFrameCreator instance and the h5 file
+        df = DataFrameCreator(config_dataframe=self._config, h5_path=paths["raw"]).df
 
-        print(f"Reading files: {len(self.missing_h5_files)} new files of {len(h5_paths)} total.")
+        # forward fill all the non-electron channels
+        df[self.fill_channels] = df[self.fill_channels].ffill()
```

Comment:
Do we need to fill them here already before saving/loading? This used to be done by ffill_lazy, no?

Reply:
Yes but then I can't drop nans in subsequent steps because filling is necessary to have correct information before dropping.
And as mentioned in other comment, dropping nans operation is really slow in dask. Likely because it modifies the df length which destroys the indexing (indexed operations are always faster).

Reply:
Then let's keep this approach (although I don't completely understand)